### ISSUE TITLE
Extern variables and functions

### ISSUE THREAD

**Issue by jakobandersen** [michaeljones/breathe#527]
See also #526.

I couldn't find any indication on variables/functions being ``extern` ` in the Doxygen XML. Is this a too-old-doxygen problem? Tested on version Doxygen version 1.8.13.

**Comment by vermeeren** [comment]
Not sure what you mean here exactly. I must admit I haven't used `extern` in quite a while and don't remember if it shows up correctly with Breathe/Sphinx.

**Comment by jakobandersen** [comment]
As far as I can see it doesn't show up, e.g., see the added example in #526.

**Comment by vermeeren** [comment]
Yeah indeed there is nothing at all in the XML regarding `extern`, I tried some searching in the Doxygen docs and issue tracker but can't seem to find anything about `extern` in context of the XML output type.

For future reference the line `extern int cache_tree_matches_traversal(struct cache_tree *, struct name_entry *ent, struct traverse_info *info);` with Doxygen 1.8.13 gives:

```xml
      <memberdef kind="function" id="c__file_8h_1a9d282a52f322bf17e11ae6e3aaa33761" prot="public" static="no" const="no" explicit="no" inline="no" virt="non-virtual">
        <type>int</type>
        <definition>int cache_tree_matches_traversal</definition>
        <argsstring>(struct cache_tree *, struct name_entry *ent, struct traverse_info *info)</argsstring>
        <name>cache_tree_matches_traversal</name>
        <param>
          <type>struct <ref refid="structcache__tree" kindref="compound">cache_tree</ref> *</type>
        </param>
        <param>
          <type>struct name_entry *</type>
          <declname>ent</declname>
        </param>
        <param>
          <type>struct traverse_info *</type>
          <declname>info</declname>
        </param>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="c_file.h" line="51" column="1"/>
      </memberdef>
```

**Comment by vermeeren** [comment]
No change when it comes to `extern` with Doxygen 1.8.18, this should be fixed upstream first I suppose.

**Comment by michaeljones** [comment]
I've opened an issue with Doxygen to try to add this information: https://github.com/doxygen/doxygen/issues/10163

I couldn't see any issue that raised it before. I'll update if there is any progress. The actual change doesn't seem to be hard but I'm not well informed and I don't know the release schedule for Doxygen either.

**Comment by michaeljones** [comment]
I made a PR and it has just been merged: https://github.com/doxygen/doxygen/pull/10164

Out in the next version of doxygen but I'm not sure when that is.

**Comment by michaeljones** [comment]
This seems to have been released in doxygen 1.9.8.
